sympy/physics/mechanics/functions.py

```diff
@@ -422,6 +447,12 @@ def find_dynamicsymbols(expression, exclude=None):
             raise TypeError("exclude kwarg must be iterable")
     else:
         exclude_set = set()
+    if isinstance(expression, Vector):
+        if reference_frame is None:
+            raise ValueError("You must provide reference_frame when passing a "
```

Comment:
I mean adding a test case for this `raise ValueError`, i.e. add test case which verifies that a `ValueError` is indeed raised when this condition is satisfied.

Reply:
@gxyd I could not find how to write a test case for checking if a ValueError is raised or not. Will it be done using assertRaises ?
Please guide. Thanks

Reply:
See this for example: https://github.com/sympy/sympy/blob/master/sympy/functions/combinatorial/tests/test_comb_numbers.py#L88

Reply:
Its showing something like this:

>>> from sympy.physics.mechanics.tests.test_functions import test_find_dynamicsymbols
>>> test_find_dynamicsymbols()Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/saloni/sympy/sympy/physics/mechanics/tests/test_functions.py", line 181, in test_find_dynamicsymbols
    raises(ValueError, lambda: find_dynamicsymbols(v))
NameError: name 'raises' is not defined

Reply:
Ofcourse you will need to import `raises`, its a sympy utility, not python's default.

Reply:
@gxyd @moorepants I guess I have addressed all the requested changes. If not do let me know. I also couldn't figure out why the build is failing. Looking forward for it to be merged.
Thanks